connectors/redis/redis.go

```diff
@@ -96,7 +96,7 @@ func NewConnector(config Config, scope metrics.Scope) dosa.Connector {
 	return &Connector{
 		client: NewRedigoClient(config.ServerSettings, scope),
 		ttl:    config.TTL,
-		stats:  scope,
+		stats:  metrics.CheckIfNilStats(scope),
```

Comment:
When would it make sense to pass a `nil` value here? Why not make`scope` required and if its value is `nil`, panic?

Reply:
We want the scope operation to work as normal even it is nil, by adding the CheckIfNilStats, we don't need to check status every time we want to use the `stats`.